debian/ubuntu-advantage-tools.postinst

```diff
@@ -205,17 +227,12 @@ EOF
 
 configure_esm() {
     rm -f $APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg  # Remove legacy esm keys
-    if [ "14.04" = "${VERSION_ID}" ]; then
-            enable_apps="false"
-    else
-            enable_apps="true"
+    if check_is_active_esm "${UBUNTU_CODENAME}"; then
+        install_esm_apt_key_and_source "infra" "$UBUNTU_CODENAME"
     fi
-    if [ "${enable_apps}" = "true" ]; then
-        if ! check_service_is_beta esm-apps; then
-            install_esm_apt_key_and_source "apps" "$UBUNTU_CODENAME"
-        fi
+    if ! check_service_is_beta esm-apps; then
+        install_esm_apt_key_and_source "apps" "$UBUNTU_CODENAME"
```

Comment:
Doe we also need to check to make sure we're not on trusty before calling `install_esm_apt_key_and_souce "apps"` ?

Reply:
Nice catch, yes. just added.